File: Facultate/Anul 2/Sem 2/MPP/Laburi/CSharpTema3/CSharpLab3/utils/Observable.cs
namespace CSharpLab3.utils;

public abstract class Observable
{
    private readonly List<IObserver> _observers = new();

    public void AddObserver(IObserver observer) => _observers.Add(observer);
    public void RemoveObserver(IObserver observer) => _observers.Remove(observer);
    public List<IObserver> GetObservers() => _observers;
    protected void NotifyObservers() => _observers.ForEach(o => o.Update());
}

File: Facultate/Anul 2/Sem 2/MPP/MamaLuiMihai/src/main/java/org/example/observer/Observable.java
package org.example.observer;

import java.util.ArrayList;
import java.util.List;

public class Observable {

    private final List<Observer> observers = new ArrayList<>();

    public void addObserver(Observer o) {
        observers.add(o);
    }

    public void removeObserver(Observer o) {
        observers.remove(o);
    }

    public void notifyObservers(){
        for (Observer o : observers) {
            o.update();
        }
    }

}
